llvm/include/llvm/Object/ELFObjectFile.h

```diff
@@ -395,7 +396,8 @@ template <class ELFT> class ELFObjectFile : public ELFObjectFileBase {
 
     for (const Elf_Shdr &Sec : *SectionsOrErr) {
       if (Sec.sh_type == ELF::SHT_ARM_ATTRIBUTES ||
-          Sec.sh_type == ELF::SHT_RISCV_ATTRIBUTES) {
+          Sec.sh_type == ELF::SHT_RISCV_ATTRIBUTES ||
```

Comment:
These values are all equal. However, we should check `getEMachine() == EM_HEXAGON` before testing a processor-specific section type

Reply:
Done